### PR TITLE
Ticket/8116 Add pagination for IP tables on post info

### PR DESCRIPTION
**Pull request by nickvergessen**
https://tracker.phpbb.com/browse/PHPBB3-8116

### CODE REVIEW

File: phpBB/includes/mcp/mcp_post.php [hunk 1]
```diff
@@ -416,6 +474,62 @@ function mcp_post_details($id, $mode, $action)
 }
 
 /**
+ * Get the number of posters for a given ip
+ *
+ * @param \phpbb\db\driver\driver_interface $db
+ * @param string $poster_ip
+ * @return int
+ */
+function phpbb_get_num_posters_for_ip(\phpbb\db\driver\driver_interface $db, $poster_ip)
```
Comment (nickvergessen) [inline phpBB/includes/mcp/mcp_post.php]: could try to deduplicate these methods, but without putting the column name directly into the query string, you loose the overview on them

File: phpBB/includes/mcp/mcp_post.php [hunk 2]
```diff
@@ -416,6 +474,62 @@ function mcp_post_details($id, $mode, $action)
...
+{
+	if ($db->get_sql_layer() == 'sqlite' || $db->get_sql_layer() == 'sqlite3')
```
Comment (bantu) [inline phpBB/includes/mcp/mcp_post.php]: This construction should be in DBAL instead.

File: phpBB/includes/mcp/mcp_post.php [hunk 3]
```diff
@@ -416,6 +474,62 @@ function mcp_post_details($id, $mode, $action)
...
+	{
+		$sql = 'SELECT COUNT(poster_id) as num_users
+								FROM (SELECT DISTINCT poster_id';
```
Comment (bantu) [inline phpBB/includes/mcp/mcp_post.php]: "(" is never closed?

File: phpBB/includes/mcp/mcp_post.php [hunk 4]
```diff
@@ -341,6 +355,23 @@ function mcp_post_details($id, $mode, $action)
 		}
 		$db->sql_freeresult($result);
 
+		if ($users == $config['posts_per_page'] || $start_users)
```
Comment (bantu) [inline phpBB/includes/mcp/mcp_post.php]: The difference between $users and $num_users is not clear.

**Comment by CHItA** [comment]
I'm closing this, as a present for this PR's second birthday.

**Pull request closed without merging**